pandas/core/generic.py

```diff
@@ -4419,6 +4423,63 @@ def sort_values(
         """
         raise AbstractMethodError(self)
 
+    def sort_index(
+        self,
+        axis=0,
+        level=None,
+        ascending: bool_t = True,
+        inplace: bool_t = False,
+        kind: str = "quicksort",
+        na_position: str = "last",
+        sort_remaining: bool_t = True,
+        ignore_index: bool_t = False,
+        key: IndexKeyFunc = None,
+    ):
+
+        inplace = validate_bool_kwarg(inplace, "inplace")
+        axis = self._get_axis_number(axis)
+        target = self._get_axis(axis)
+
+        indexer = get_indexer_indexer(
+            target, level, ascending, kind, na_position, sort_remaining, key
+        )
+
+        if indexer is None:
+            if inplace:
+                return
+            else:
+                return self.copy()
+
+        if isinstance(self, ABCDataFrame):
+            baxis = self._get_block_manager_axis(axis)
+            new_data = self._mgr.take(indexer, axis=baxis, verify=False)
+
+            # reconstruct axis if needed
+            new_data.axes[baxis] = new_data.axes[baxis]._sort_levels_monotonic()
+
+            if ignore_index:
+                new_data.axes[1] = ibase.default_index(len(indexer))
+
+            result = self._constructor(new_data)
+
+        else:
+            indexer = ensure_platform_int(indexer)
+            new_index = self.index.take(indexer)
+            new_index = new_index._sort_levels_monotonic()
+
+            new_values = self._values.take(indexer)
```

Comment:
doesn't the path for the DataFrame work for both Series/DataFrame? (it *should* and if not that's something that should fix (give it a try)